### PR TITLE
Unit test/data source

### PR DESCRIPTION
**Pull request by vulh-infragistics**

### CODE REVIEW

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace Reveal.Sdk.Dom.Tests.Data
     public class DataSourceFixture
     {
         [Fact]
-        public void DataSource_DefaultConstructor_SetsSchemaTypeName()
+        public void Constructor_SetDefaultDataSource_WithoutArguments()
         {
             // Arrange
             var dataSource = new DataSource();
```
[the remark this answers is not in the document]
Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs:13]: Updated

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public void DataSource_DefaultConstructor_SetsSchemaTypeName()
         }
 
         [Fact]
-        public void DataSource_DefaultConstructor_SetsProperties()
+        public void Constructor_SetDefaultProperties_WithoutArguments()
         {
             // Arrange
             var dataSource = new DataSource();
```
[the remark this answers is not in the document]
Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs:25]: Updated

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs [hunk 3]
```diff
@@ -99,7 +106,7 @@ public void DataSource_Equals_SameId_ReturnsTrue()
         }
 
         [Fact]
-        public void DataSource_GetHashCode_ReturnsConsistentValue()
+        public void GetHashCode_ReturnsConsistentValue_ForTheSameDataSource()
```
[the remark this answers is not in the document]
Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs:140]: I have added a unit test to check HashCode function generates different hash code with different paramters. 
I think the current unit test already includes the case of getting hashcode with same Id

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs [hunk 4]
```diff
@@ -111,5 +118,18 @@ public void DataSource_GetHashCode_ReturnsConsistentValue()
             // Assert
             Assert.Equal(hashCode1, hashCode2);
         }
+
+        [Fact]
+        public void CheckEqual_ReturnFalse_WhenComparedWithNull()
```
[the remark this answers is not in the document]
Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSourceFixture.cs]: Removed

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItemFixture.cs [hunk 5]
```diff
@@ -10,61 +12,234 @@ namespace Reveal.Sdk.Dom.Tests.Data
     public class DataSourceItemFixture
     {
         [Fact]
-        public void DataSourceItem_Id_DefaultValue()
+        public void Constructor_GeneratesNotNullObject_WithoutArguments()
         {
             // Arrange
             var dataSourceItem = new DataSourceItem();
```
[the remark this answers is not in the document]
Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSourceItemFixture.cs]: Updated

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItemFixture.cs [hunk 6]
```diff
@@ -112,31 +288,34 @@ public void DataSourceItem_Join_AddsJoinTableUsingJoinConditions()
...
+            Assert.Equal($"{aliasName}.[{rightJoinFieldName}]", dataSourceItem.JoinTables[0].JoinConditions[0].RightFieldName);
         }
 
         [Theory]
         [InlineData("FieldName", "[FieldName]")]
         [InlineData("[FieldName]", "[FieldName]")]
         [InlineData("A.FieldName", "A.[FieldName]")]
         [InlineData("A.[FieldName]", "A.[FieldName]")]
-        public void ValidateLeftJoinFieldName_ShouldReturnFormattedFieldName(string fieldName, string expected)
+        public void ValidateLeftJoinFieldName_ReturnsFieldName_WithLeftFieldName/**/(string fieldName, string expected)
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItemFixture.cs:318]: Remove `/**/`

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

**Review by hainv-ohio — APPROVED** [review]

**Pull request merged**